src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -725,4 +775,328 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
+					cellClasses += ' missing-value';
+				}
+				return '<span title="' + valueToDisplay + '" class="' + cellClasses + '">' + valueToDisplay + '</span>';
+			}
+			return undefined;
+		};
+	}
+
+	handleChanges(changes: { [propName: string]: any }): void {
+		let columnDefinitionChanges = changes['columnDefinitions'];
+		let activeCell = this.tables[0] ? this.tables[0].grid.getActiveCell() : undefined;
+		let hasGridStructureChanges = false;
+		let wasEditing = this.tables[0] ? !!this.tables[0].grid.getCellEditor() : false;
+
+		if (columnDefinitionChanges && !equals(columnDefinitionChanges.previousValue, columnDefinitionChanges.currentValue)) {
+			if (!this.tables[0]) {
+				this.createNewTable();
+			} else {
+				this.tables[0].grid.resetActiveCell();
+				this.tables[0].grid.setColumns(this.dataSet.columnDefinitions);
+			}
+			hasGridStructureChanges = true;
+
+			if (!columnDefinitionChanges.currentValue || columnDefinitionChanges.currentValue.length === 0) {
+				activeCell = undefined;
+			}
+			if (activeCell) {
+				let columnThatContainedActiveCell = columnDefinitionChanges.previousValue[Math.max(activeCell.cell - 1, 0)];
+				let newActiveColumnIndex = columnThatContainedActiveCell
+					? columnDefinitionChanges.currentValue.findIndex(c => c.id === columnThatContainedActiveCell.id)
+					: -1;
+				activeCell.cell = newActiveColumnIndex !== -1 ? newActiveColumnIndex + 1 : 0;
+			}
+		}
+
+		if (changes['dataRows']
+			|| (changes['highlightedCells'] && !equals(changes['highlightedCells'].currentValue, changes['highlightedCells'].previousValue))
+			|| (changes['blurredColumns'] && !equals(changes['blurredColumns'].currentValue, changes['blurredColumns'].previousValue))
+			|| (changes['columnsLoading'] && !equals(changes['columnsLoading'].currentValue, changes['columnsLoading'].previousValue))) {
+			this.setCallbackOnDataRowsChanged();
+			this.tables[0].grid.updateRowCount();
+			this.tables[0].grid.setColumns(this.tables[0].grid.getColumns());
+			this.tables[0].grid.invalidateAllRows();
+			this.tables[0].grid.render();
+			hasGridStructureChanges = true;
+		}
+
+		if (hasGridStructureChanges) {
+			if (activeCell) {
+				this.tables[0].grid.setActiveCell(activeCell.row, activeCell.cell);
+			} else {
+				this.tables[0].grid.resetActiveCell();
+			}
+		}
+
+		if (wasEditing && hasGridStructureChanges) {
+			this.tables[0].grid.editActiveCell(this.tables[0].grid.getCellEditor());
+		}
+	}
+
+	private setCallbackOnDataRowsChanged(): void {
+		if (this.dataSet.dataRows) {
+			// We must wait until we get the first set of dataRows before we enable editing or slickgrid will complain
+
+			//line here may be redundant.
+			if (this.enableEditing) {
+				this.enterEditSession();
+			}
+
+			this.dataSet.dataRows.setCollectionChangedCallback((startIndex: number, count: number) => {
+				this.renderGridDataRowsRange(startIndex, count);
+			});
+		}
+	}
+
+	// Enables editing on the grid
+	public enterEditSession(): void {
```

Comment:
typically if a method only calls into another method, it is often better to remove that extra method and call the final intended target method directly.